sdk/eventhub/Directory.Build.props

```diff
@@ -10,6 +10,7 @@
   <PropertyGroup Condition="$(MSBuildProjectName.StartsWith('Azure.')) and $(MSBuildProjectName.Contains('.Samples'))">
     <IsSamplesProject>true</IsSamplesProject>
     <IsTestSupportProject>true</IsTestSupportProject>
+    <IsOptOutOfGlobalTestTimeout>true</IsOptOutOfGlobalTestTimeout>
   </PropertyGroup>
```

Comment:
```suggestion
  <PropertyGroup Condition="$(MSBuildProjectName.StartsWith('Azure.')) and $(MSBuildProjectName.Contains('.Samples'))">
    <IsSamplesProject>true</IsSamplesProject>
    <IsTestSupportProject>true</IsTestSupportProject>
  </PropertyGroup>
  
  <!-- 
    OptOut of the global NUnit test timeout for all Event Hubs libraries.
    
    The approach that NUnit uses to measure timing for tests does not play well with parallel execution
    and highly asynchronous code.  Many of the Event Hubs tests run against non-deterministic background
    operations where timing is longer than normal unit tests and the exact timing is difficult to pin down.
    
    Tests currently use an environment variable with cancellation-based timeout control, which allows for
    more generous timeouts in the CI environments, where resources are often more constrained and 
    continuations take longer to be scheduled.    
  -->
  <PropertyGroup>
        <IsOptOutOfGlobalTestTimeout>true</IsOptOutOfGlobalTestTimeout>
  </PropertyGroup>
```

We probably want to slide this out into its own group;  the conditionals on this one specifically target sample projects.